src/webview/markdownEngine.ts

```diff
@@ -0,0 +1,99 @@
+import * as hljs from "highlight.js";
+import * as MarkdownIt from "markdown-it";
+import * as path from "path";
+import * as vscode from "vscode";
+import { leetCodeChannel } from "../leetCodeChannel";
+
+export class MarkdownEngine {
+
+    public readonly engine: MarkdownIt;
+    public readonly extRoot: string; // root path of vscode built-in markdown extension
+
+    public constructor() {
+        this.engine = this.initEngine();
+        this.extRoot = path.join(vscode.env.appRoot, "extensions", "markdown-language-features");
+    }
+
+    public get localResourceRoots(): vscode.Uri[] {
+        return [vscode.Uri.file(path.join(this.extRoot, "media"))];
+    }
+
+    public get styles(): vscode.Uri[] {
+        try {
+            const stylePaths: string[] = require(path.join(this.extRoot, "package.json"))["contributes"]["markdown.previewStyles"];
+            return stylePaths.map((p: string) => vscode.Uri.file(path.join(this.extRoot, p)).with({ scheme: "vscode-resource" }));
+        } catch (error) {
+            leetCodeChannel.appendLine("[Error] Fail to load built-in markdown style file.");
+            return [];
+        }
+    }
+
+    public get options(): MarkdownIt.Options {
+        return (this.engine as any).options;
+    }
+
+    public getStylesHTML(): string {
+        return this.styles.map((style: vscode.Uri) => `<link rel="stylesheet" type="text/css" href="${style.toString()}">`).join("\n");
+    }
+
+    public render(md: string, env?: any): string {
+        return this.engine.render(md, env);
+    }
+
+    private initEngine(): MarkdownIt {
+        const md: MarkdownIt = new MarkdownIt({
+            linkify: true,
+            typographer: true,
+            highlight: (code: string, lang?: string): string => {
+                if (lang && ["tsx", "typescriptreact"].indexOf(lang.toLocaleLowerCase()) >= 0) {
```

Comment:
I think using `switch-case` block will be better than the multiple `if` blocks

Reply:
This is the routine adopted by built-in markdown extension's [engine](https://github.com/Microsoft/vscode/blob/master/extensions/markdown-language-features/src/markdownEngine.ts#L292). I agree `switch case` is a better choice, and will adopt it in next commit.

Reply:
How about this way?
```javascript
lang = lang && ({
    mysql: "sql",
    json5: "json",
    python3: "python",
})[lang.toLowerCase()];
```
"tsx" and "typescriptreact" is ruled out, and "mysql" is taken account.

Besides, a language's name is always English, so no need to consider locale mapping.